### PR TITLE
Update mypy and make `AbstractSaveable.save` return `Self` type

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: etna/core/mixins.py [hunk 1]
```diff
@@ -163,7 +166,7 @@ def get_etna_version() -> Tuple[int, int, int]:
         return result
 
 
-class SaveMixin:
+class SaveMixin(AbstractSaveable):
```
Comment (Mr-Geekman) [inline etna/core/mixins.py:169]: It won't work without this somehow.

**Comment by GrozniyToaster** [comment]
It looks like it won't work without inheritance.

Mypy is bad at checking abstract class signatures, you can try with Protocols

**Review by GrozniyToaster — COMMENTED** [review]
looks nice

**Review by GrozniyToaster — APPROVED** [review]
looks nice v2

**Pull request merged**